src/core/stdc/time.d

```diff
@@ -15,138 +15,22 @@
 
 module core.stdc.time;
 
-private import core.stdc.config;
+version (CRuntime_Abstract)
+    public import platform_dependent.stdc.time;
```

Comment:
Need something else here perhaps.

Reply:
if you about
```D
private import core.stdc.config;
```
it moved to 27 line

Reply:
Please explain more

Reply:
Neither the version exists in the spec nor the module exists in druntime.

Reply:
Yes, this is made intentionally. This version must be used if you want to connect an arbitrary external platform implementation.

Currently it is not used in *.mak scripts. It makes no sense to add it because some CI test implementation code for it does not exist yet.

Reply:
I'd just leave such cases to hit the static assert. Don't bother trying to cater for it.

Reply:
@ibuclaw i.e., should try to compile `import platform_dependent.*` and if it is not compiles then throw static assertion?

Reply:
> @ibuclaw i.e., should try to compile `import platform_dependent.*` and if it is not compiles then throw static assertion?

Hi, sorry, I dropped the ball here.

I think for now we need not bother with handling unknown platforms in druntime.

If we ever need such machinery, better to do it all in one go. So please just use `static assert(false, "unsupported platform");` to be consistent with the rest of druntime.

Reply:
> I think for now we need not bother with handling unknown platforms in druntime.

Not "unknown" but "abstract".

> If we ever need such machinery, better to do it all in one go.

It is already needed! Al these MCUs, thousands of configarations! (like: Bare metal ARM, FreeRTOS ARM, bare metal RISC-V, FreeRTOS RISC-V, FreeRTOS RISC-V with another Memory Manager, etc.)
Also as a side effect we get rid of libc.

Reply:
In other words, my proposal is to make possible to replace of the low-level calls used by by druntime such as memory allocation or thread creation without changing druntime code.

Reply:
> In other words, my proposal is to make possible to replace of the low-level calls used by druntime such as memory allocation or thread creation without changing druntime code.

Then we can do that as a phase-2 step, and add all ~30 places in druntime in one PR. :-) 

First, let's just split up core.time, to keep refactoring and actual changes separated.

Reply:
> Then we can do that as a phase-2 step, and add all ~30 places in druntime in one PR. :-)

Ok, but first we need to reach an agreement that this is exactly the way to add an "abstract" platform that will be merged. Otherwise this huge amount of work will be wasted.

> First, let's just split up core.time, to keep refactoring and actual changes separated.

Now I think this is not the best candidate for the first change. It is too huge. Maybe I find something smaller.

Reply:
> > First, let's just split up core.time, to keep refactoring and actual changes separated.
> 
> Why you talk about  core.time?
> If it isn't typo now I think this is not the best candidate for the first change. It is too huge. Maybe I find something smaller.

I did indeed mean core.stdc.time.

Furthermore CRuntime_Abstract is an  undocumented, unimplented, unapproved version condition; and it ignores an already existing and documented version condition made precisely for bare metal targets.

Reply:
So I reiterate, let's remove these two lines for now, and think more broadly about how druntime could support free-standing environments in the future.

Reply:
Ok, I removed these two lines